File: 조준장/level2/삼각 달팽이.py
def solution(n):
    answer = []
    graph = []

    graph = [[0 for j in range(i)] for i in range(1, n+1)] # 삼각형 구조 만들기

    x = -1
    y = 0
    num = 0
    for v in range(n):
        for s in range(v, n):
            if v % 3 == 0:
                x += 1
                
            elif v % 3 == 1:
                y += 1
                
            else:
                x -= 1
                y -= 1
                
            num += 1
            graph[x][y] = num
                
    for g in graph:
        answer += g
        
    return answer

File: 이훈일/Programmers/Level1/삼각 달팽이.java
class Solution {
    public int[] solution(int n) {
        int[] dix = new int[]{1,0,-1};
        int[] diy = new int[]{0,1,-1};

        int[][] arr = new int[n][];
        for(int i = 0; i < n; i++) {
            arr[i] = new int[i+1];
        }

        int x = -1;
        int y = 0;
        int num = 0;
        for(int i = 0; i < n; i++) {
            for(int j = 0; j < n-i; j++) {
                x += dix[i%3];
                y += diy[i%3];
                arr[x][y] = ++num;
            }
        }

        int[] answer = new int[num];
        int count = 0;
        for(int i = 0; i < arr.length; i++) {
            for(int j = 0; j < arr[i].length; j++) {
                answer[count] = arr[i][j];
                count++;
            }
        }

        return answer;
    }
}
